topics/dev/tutorials/interactive-tools/tutorial.md

```diff
@@ -525,6 +530,535 @@ our new Docker container as a Galaxy tool.
+
+> <tip-title>A brief primer on Docker</tip-title>
+> Docker allows an entire application context to be containerized. A typical web application consists of an operating system, installed dependancies, web server configuration, database configuration and, of course, the codebase of the software itself. A Docker container can encapsulate all of these components in a single "image", which can be run on any machine with Docker installed.
+>
+> **Essentials of Docker:**
+>
+> 1. Write an image recipe as a Dockerfile. This single file selects an OS, installs software, pulls code repositories and copies files from the host machine (your computer).
+> 2. Build the image from your recipe:
+>
+>    `docker build -t <image_name> .`
+> 3. View existing images with
+>
+>    `docker image list`
+> 4. Run a container with a specified command:
+>
+>    `docker run <image_name> <command>`
+> 5. View running containers:
+>
+>    `docker ps`
+> 6. Stop a running container:
+>
+>    `docker stop <container_name>`
+> 7. Remove a stopped container:
+>
+>    `docker container rm <container_name>`
+> 8. Remove an image:
+>
+>    `docker image rm <container_name>`
+{: .tip}
+
+
+Let's check out [the Dockerfile](https://github.com/usegalaxy-eu/docker-copernicus-notebooks/blob/main/Dockerfile) that we'll use to containerize our Jupyterlab.
+
+This container recipe can be used to build a Docker image which can be pushed to a
+container registry in the cloud, ready for consumption by our Galaxy instance:
+
+```dockerfile
+# Jupyter container used for Galaxy copernicus notebooks (+other kernels) Integration
+
+# from 5th March 2021
+FROM jupyter/datascience-notebook:python-3.10
+
+MAINTAINER Björn A. Grüning, bjoern.gruening@gmail.com
+
+ENV DEBIAN_FRONTEND noninteractive
+USER root
+
+RUN apt-get -qq update && \
+    apt-get install -y wget unzip net-tools procps && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Set channels to (defaults) > bioconda > conda-forge
+RUN conda config --add channels conda-forge && \
+    conda config --add channels bioconda
+    #conda config --add channels defaults
+RUN pip install --upgrade pip
+RUN pip install --no-cache-dir bioblend galaxy-ie-helpers
+
+ENV JUPYTER /opt/conda/bin/jupyter
+ENV PYTHON /opt/conda/bin/python
+ENV LD_LIBRARY_PATH /opt/conda/lib/
+
+# Python packages
+RUN conda config --add channels conda-forge && \
+    conda config --add channels bioconda && \
+    conda install --yes --quiet \
+    bash_kernel \
+    ansible-kernel \
+    bioblend galaxy-ie-helpers \
+```
+With those packages installed you have the bases to have a functionnal environement in your jupyterlab AND be able to link your jupyterlab to the Galaxy history with the pacakage `bioblend galaxy-ie-helpers`.
```

Comment:
```suggestion
With those packages installed, you have the bases to have a functional environment in your JupyterLab AND be able to link your JupyterLab to the Galaxy history with the package `bioblend galaxy-ie-helpers`.
```

Reply:
the package is galaxy-ie-helpers, it uses under the hood bioblend, but I guess we can ignore this here